v3/tags_data.go

```diff
@@ -909,6 +909,25 @@ IFD:
 - id: 0xc74e
   name: OpcodeList3
   type_name: UNDEFINED
+# This tag may be used to specify the size of raster pixel spacing in the model space units, when the raster space can be embedded in the model space coordinate system without rotation, and consists of the following 3 values:
+#    ModelPixelScaleTag = (ScaleX, ScaleY, ScaleZ)
+# where ScaleX and ScaleY give the horizontal and vertical spacing of raster pixels. The ScaleZ is primarily used to map the pixel value of a digital elevation model into the correct Z-scale, and so for most other purposes this value should be zero (since most model spaces are 2-D, with Z=0).
+# Source: http://geotiff.maptools.org/spec/geotiff2.6.html#2.6.1
+- id: 0x830e
+  name: ModelPixelScaleTag
+  type_name: RATIONAL
```

Comment:
The GeoTIFF tags are documented to be *signed doubles*. You should use the `SRATIONAL` type (versus the RATIONAL type) in _go-exif_, no?

I'm also concerned that some of the GeoTIFF tag definitions say "DOUBLE (IEEE Double precision)" and others simply say "DOUBLE", and there are several occurrences of both. Can you do research on the difference and share here?

```
ModelPixelScaleTag:
      Tag = 33550
      Type = DOUBLE (IEEE Double precision)
      N = 3
      Owner: SoftDesk

ModelTiepointTag:
      Tag = 33922 (8482.H) 
      Type = DOUBLE (IEEE Double precision)
      N = 6*K,  K = number of tiepoints
      Alias: GeoreferenceTag
      Owner: Intergraph

ModelTransformationTag
      Tag  =  34264  (85D8.H) 
      Type =  DOUBLE    
      N    =  16
      Owner: JPL Cartographic Applications Group

IntergraphMatrixTag
      Tag  =  33920  (8480.H) 
      Type =  DOUBLE    
      N    =  17 (Intergraph implementation) or 16 (GeoTIFF 0.2 impl.)
      Owner: Intergraph
```

Reply:
> The GeoTIFF tags are documented to be signed doubles. You should use the SRATIONAL type (versus the RATIONAL type) in go-exif, no?

I wasn't aware about the difference between `RATIONAL` and `SRATIONAL`, I'll replace by `SRATIONAL`. `ModelPixelScaleTag` can have negative values just like `ModelTiepointTag` with some coordinate values.

>  I'm also concerned that some of the GeoTIFF tag definitions say "DOUBLE (IEEE Double precision)" and others simply say "DOUBLE", and there are several occurrences of both. Can you do research on the difference and share here?

![image](https://user-images.githubusercontent.com/6754291/100475207-4a66bb00-30c1-11eb-907b-0e59080bc346.png)

Looking at [DGIWG specifications](https://portal.dgiwg.org/files/?artifact_id=68102&format=doc) seems there's no difference between "DOUBLE (IEEE Double precision)" and "DOUBLE", should we add the support to `DOUBLE` as a type for go-exif? Looking at exiftool tag list, there's another tags using double

Reply:
The EXIF standard doesn't have the concept of float/double/decimals. It implements rationals, instead. Can you look into the wire-format of a double in GeoTIFF and also find the tag-type ID that indicates a DOUBLE?

Reply:
> Can you look into the wire-format of a double in GeoTIFF

Sorry, I don't get what you mean here, do you want value examples? (I have some for `ModelPixelScaleTag` and `ModelTiepointTag`)

> find the tag-type ID that indicates a DOUBLE?

I don't have any idea about where to search about tag-type IDs, it should be at some specification?

Reply:
The values are encoded down to bytes in the EXIF stream. The types of those values are encoded as SHORTs. So, you can't just _call something_ a type if it doesn't actually encode that way on disk (so that the semantics we'll use to decode those bytes will recover the correct value). Knowing the tag-ID will either a) tell us that it's actually using one of the current types and no work needs to be done (unlikely), or b) will give us a clue to help search for implementation details.

Reply:
It's on page 16, here: https://www.itu.int/itudoc/itu-t/com16/tiff-fx/docs/tiff6.pdf

![image](https://user-images.githubusercontent.com/715626/100648385-dc173800-330e-11eb-9321-84b9c4657bac.png)

EXIF, which inherits from TIFF, doesn't specify FLOAT/DOUBLE. However, we can add it since it looks like the tag-type is standardish.

It looks like Go should properly handle the wire-level encoding (https://golang.org/pkg/encoding/binary -> https://developers.google.com/protocol-buffers/docs/encoding -> https://stackoverflow.com/questions/7248950/how-cross-platform-is-googles-protocol-buffers-handling-of-floating-point-type), so all we need to do is just update:

https://github.com/dsoprea/go-exif/blob/56058635d0feba16325f88e9447666f746e9d39a/v3/common/type.go#L37

https://github.com/dsoprea/go-exif/blob/56058635d0feba16325f88e9447666f746e9d39a/v3/common/type.go#L156

https://github.com/dsoprea/go-exif/blob/56058635d0feba16325f88e9447666f746e9d39a/v3/common/type.go#L277

https://github.com/dsoprea/go-exif/blob/56058635d0feba16325f88e9447666f746e9d39a/v3/common/type.go#L359

https://github.com/dsoprea/go-exif/blob/56058635d0feba16325f88e9447666f746e9d39a/v3/common/value_context.go#L365

https://github.com/dsoprea/go-exif/blob/56058635d0feba16325f88e9447666f746e9d39a/v3/common/value_encoder.go#L186

We should do both float and double. It should be fairly simple. A little bit of code in each location and some unit-tests, and there is extensive precedent from the implementations of the current types.

Do you want to give that a shot (in a separate PR)?

Reply:
This also explains "type 8", which has been used, albeit not very much, but I didn't previously know where it came from.

https://github.com/dsoprea/go-exif/blob/56058635d0feba16325f88e9447666f746e9d39a/v3/common/type.go#L58

Reply:
Can you please send me your sample image, directly? dustin@randomingenuity.com

Reply:
I've sent an e-mail and I'll try to implement the changes in a separate PR as requested (probably as a draft first)